src/families/stellar/js-signOperation.ts

```diff
@@ -7,21 +7,20 @@ import { open, close } from "../../hw";
 import type { Transaction } from "./types";
 import { buildTransaction } from "./js-buildTransaction";
 import { fetchSequence } from "./api";
+import { getAmountValue } from "./helpers/getAmountValue";
 
 const buildOptimisticOperation = async (
   account: Account,
   transaction: Transaction
 ): Promise<Operation> => {
   const transactionSequenceNumber = await fetchSequence(account);
   const fees = transaction.fees ?? new BigNumber(0);
+
   const operation: Operation = {
     id: `${account.id}--OUT`,
     hash: "",
     type: "OUT",
-    value:
-      transaction.useAllAmount && transaction.networkInfo
-        ? account.balance.minus(transaction.networkInfo.baseReserve).minus(fees)
-        : transaction.amount.plus(fees),
+    value: getAmountValue(account, transaction, fees),
```

Comment:
It seems this changes the computation in the native case?

Reply:
The native amount is calculated differently, because of the required reserve and fees. Non-native asset balances don’t have these requirements (for non-native transactions we need to check asset amount + native (XLM) amount to make sure there is enough to cover the transaction). Was that your question?

Reply:
My concern is mainly avoiding regressions, I was just asking about this change:
```diff
-     value:
-      transaction.useAllAmount && transaction.networkInfo
-        ? account.balance.minus(transaction.networkInfo.baseReserve).minus(fees)
-        : transaction.amount.plus(fees)

compared to getAmountValue:
+   return transaction.useAllAmount && transaction.networkInfo
+    ? BigNumber.max(account.spendableBalance.minus(fees), 0)
+   : transaction.amount;
```

Reply:
I don't think we need to add the fee to the amount for XLM because it's submitted separately. If I'm sending someone 1 XLM and set the fee 0.5 XLM, the total amount I'm sending should still be 1 XLM, and max fee paid will be 0.5 XLM (could be less). Is that your concern?